client/ayon_resolve/plugins/create/create_editorial_package.py

```diff
@@ -1,88 +1,155 @@
-                "taskName": task["name"],
-            })
-
         if not current_timeline:
-            raise RuntimeError("Make sure to have an active current timeline.")
+            raise CreatorError("Make sure to have an active current timeline.")
 
         timeline_media_pool_item = lib.get_timeline_media_pool_item(
             current_timeline
         )
 
-        publish_data = deepcopy(self.data)
-        # add publish data for streamline publishing
-        publish_data["publish"] = get_editorial_publish_data(
-            folder_path=folder["path"],
-            product_name=self.data["productName"],
+        tag_metadata = {
+            "publish": deepcopy(instance_data),
+        }
+        tag_metadata["publish"].update(
+            get_editorial_publish_data(
+                folder_path=instance_data["folderPath"],
+                product_name=product_name
+            )
         )
+        tag_metadata["publish"]["label"] = current_timeline.GetName()
 
         timeline_media_pool_item.SetMetadata(
-            lib.pype_tag_name, json.dumps(publish_data)
+            constants.AYON_TAG_NAME, json.dumps(tag_metadata)
         )
 
-
-def ask_for_context(
-    project_name, folder_id
-):
-    """Ask for context to create Editorial Package."""
-    controller = ContextDialogController()
-    window = ContextDialog(controller=controller)
-    controller.set_expected_selection(
-        project_name, folder_id)
-    window.exec_()
-
-    return controller.get_selected_context()
+        new_instance = CreatedInstance(
+            self.product_type,
+            product_name,
+            tag_metadata["publish"],
+            self,
+        )
+        new_instance.transient_data["timeline_pool_item"] = (
+            timeline_media_pool_item)
+        self._add_instance_to_context(new_instance)
+
+    def collect_instances(self):
+        """Collect all created instances from current timeline."""
+        for media_pool_item in lib.iter_all_media_pool_clips():
+            data = media_pool_item.GetMetadata(constants.AYON_TAG_NAME)
+            if not data:
+                continue
+
+            try:
+                data = json.loads(data)
+            except json.JSONDecodeError:
+                self.log.warning(
+                    f"Failed to parse json data from media pool item: "
+                    f"{media_pool_item.GetName()}"
+                )
+                continue
+
+            # exclude all which are not productType editorial_pkg
+            if (
+                data.get("publish", {}).get("productType") != self.product_type
+            ):
+                continue
+
+            publish_data = data["publish"]
+
+            # TODO: backward compatibility for legacy workflow instances
+            # add label into instance data in case it is missing in publish
+            # data
+            if "label" not in publish_data:
+                publish_data["label"] = media_pool_item.GetName()
```

Comment:
Why is `label` data so important that we require it in `instance.data`? Can't we just ignore it not existing? (And maybe even more forward with not setting it now, or?)

Reply:
So I understood from Robin this is purely cosmetic - we just want to display the timeline name in the publisher instead of the product name.

Hence, I'd just ALWAYS set this and disregard it from whatever is imprintend on the instance.

If someone renames the timeline, we want it display the new name - not whatever was once imprinted to it on the past.